088-rebalance-progress-status.md

```diff
@@ -0,0 +1,421 @@
+  - `executorState`: JSON object from previous update.
+- `NotReady`
+  - `estimatedTimeToCompletion`: N/A
+  - `completedDataMovementPercentage`: The value of `completedDataMovementPercentage` from previous update.
+  - `executorState`: JSON object from previous update.
+- `Ready`:
+  - `estimatedTimeToCompletion`: 0
+  - `completedDataMovementPercentage`: 100
+  - `executorState`: Empty JSON object.
+
+All the information required for the Cluster Operator to estimate the values of `estimatedTimeToCompletionInMinutes` and `completedDataMovementPercentage` fields can be derived from the Cruise Control server configurations and the [/kafkacruisecontrol/state?substates=executor](https://github.com/linkedin/cruise-control/wiki/REST-APIs#query-the-state-of-cruise-control) REST API endpoint.
+However, the actual formulas used to produce values for these fields depend on the state of the `KafkaRebalance` resource.
+Checkout the example in the [Field: executorState](#field-executorstate) section to see where the fields used in the formulas below come from.
+
+### Field: `estimatedTimeToCompletionInMinutes`
+
+The estimated time it will take in minutes for a rebalance to complete.
+
+The formulas used to calculate the field value differ for each applicable `KafkaRebalance` state:
+
+#### State: `ProposalReady`
+
+**Estimation for inter-broker rebalance:**
+
+This estimate will be a theoretical minimum derived from Cruise Control capacity and throttle configurations.
+This means that the cluster rebalance would take at least the estimated amount of time to complete.
+
+$$\text{maxPartitionMovements}_{[1]} = \min(\text{numberOfBrokers} \times \text{num.concurrent.partition.movements.per.broker}),\text{max.num.cluster.partition.movements})$$
+
+$$\text{bandwidth}_{[2]} = \min(\text{networkCapacity}, \text{replication.throttle})$$
+
+$$\text{throughput}_{[3]} = \text{maxPartitionMovements} \times \text{bandwidth}$$
+
+$$\text{estimatedTimeToCompletionInMinutes} = \frac{\text{dataToMoveMB}}{\text{throughput}}$$
+
+Notes:
+- [1] The maximum number of partition movements given Cruise Control partition movement capacity.
+
+- [2] The network bandwidth given Cruise Control bandwidth throttle.
+
+- [3] The throughput given the max allowed number of partition movements and network bandwidth.
+
+**Estimation for intra-broker rebalance:**
+
+It is challenging to provide an accurate estimate for intra-broker rebalances without an estimate for disk read/write throughput and getting disk throughput is non-trivial for Strimzi.
+Since we cannot accurately estimate `estimatedTimeToCompletionInMinutes` without knowing the disk throughput, we set `estimatedTimeToCompletionInMinutes` to `N/A`.
+
+$$\text{maxPartitionMovements}_{[1]} = \min\left(\text{numberOfBrokers} \times \text{num.concurrent.intra.broker.partition.movements.per.broker}),\text{max.num.cluster.movements}\right)$$
```

Comment:
Similar here to previous comments, you're missing a bracket in the formula, but I'm also not clear what `maxPartitionMovements` is really representing